PWGHF/Tasks/taskD0parametrizedPID.cxx

```diff
@@ -113,13 +118,43 @@ struct TaskD0parametrizedPIDMC {
       //if (ncontributor<=centralitySelectionMin && ncontributor>centralitySelectionMax) {
       //  continue;
       //}
+      Float_t maxFiducialY = 0.8;
+      Float_t minFiducialY = -0.8;
       if (std::abs(particle.flagMCMatchGen()) == 1 << DecayType::D0ToPiK) {
         if (std::abs(RecoDecay::Y(array{particle.px(), particle.py(), particle.pz()}, RecoDecay::getMassPDG(particle.pdgCode()))) > 4.0) {
           continue;
         }
         auto ptGen = particle.pt();
         auto yGen = RecoDecay::Y(array{particle.px(), particle.py(), particle.pz()}, RecoDecay::getMassPDG(particle.pdgCode()));
-        registry.fill(HIST("hMassGen"), ptGen, std::abs(yGen));
+        registry.fill(HIST("hGenPtVsY"), ptGen, std::abs(yGen));
+        if (ptGen < 5.0) {
+          maxFiducialY = -0.2 / 15 * ptGen * ptGen + 1.9 / 15 * ptGen + 0.5;
+          minFiducialY = 0.2 / 15 * ptGen * ptGen - 1.9 / 15 * ptGen - 0.5;
+        }
+        float etaProngs[2];
+        auto daught0 = particlesMC.iteratorAt(particle.daughter0Id());
+        auto daught1 = particlesMC.iteratorAt(particle.daughter1Id());
+        etaProngs[0] = std::abs(daught0.eta());
+        etaProngs[1] = std::abs(daught1.eta());
```

Comment:
Why don't you just do this?
```
        auto etaProng0 = std::abs(particle.daughter0().eta()));
```
I am not sure if you would need `daughter_as` here though.